### PR TITLE
Adds feature comparison table to README

### PR DESCRIPTION
**Pull request by skrawcz**
So that people can get a quick sense of how Hamilton stacks up against common things its compared to.

## Changes
 - Adds to README

## How I tested this
 - Locally

## Checklist

- [ ] PR has an informative and human-readable title (this will be pulled into the release notes)
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future TODOs are captured in comments
- [ ] Project documentation has been updated if adding/changing functionality.

## what it looks like:
Screen Shot 2023-02-24 at 10 09 31 PM

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,6 +32,21 @@ before forking.
 
 For the backstory on how Hamilton came about, see the original Stitch Fix [blog post!](https://multithreaded.stitchfix.com/blog/2021/10/14/functions-dags-hamilton/).
 
+# Feature comparison
+Here are common things that Hamilton is compared to, and how Hamilton compares to them.
+
+| Feature                                   | Hamilton | Macro orchestration systems like Airflow | Feast | DBT | Pandas |
```
Comment (elijahbenizzy) [inline README.md:38]: Are prefect, dagster, etc... all in `Macro orchestration systems like airflow`?

Comment (elijahbenizzy) [inline README.md:38]: Pyspark/dask could be two others that people compare it to

**Review by elijahbenizzy — COMMENTED** [review]
Looks good, a few minor points

File: README.md (hunk 1 above)
Comment (skrawcz) [reply README.md:38]: yes.

Comment (skrawcz) [reply README.md:38]: yep I'll replace pandas with dask.

**Review by elijahbenizzy — APPROVED** [review]

**Pull request merged**